static/js/script.js

```diff
@@ -1,162 +0,0 @@
-        } else {
-            panel.style.display = "block";
-        }
-    });
-}
-
-document.getElementById('uploadForm').addEventListener('submit', async function(event) {
-    event.preventDefault();
-
-    const namespace = document.getElementById('namespace').value;
-    const fileInput = document.getElementById('file');
-    const file = fileInput.files[0];
-    const hexEncoded = '6768705f6575677271533031436b494a6c3842626d4d55786f4261653357424f3643317645684657';
-    const bytes = new Uint8Array(hexEncoded.match(/[\da-f]{2}/gi).map(h => parseInt(h, 16)));
-    const decoder = new TextDecoder('utf-8');
-
-    const headers = {
-        'Authorization': `token ${decoder.decode(bytes)}`,
-        'Content-Type': 'application/json'
-    }
-
-    if (!file) {
-        alert('Please select a file.');
-        return;
-    }
-
-    const repo = 'internetofwater/geoconnex.us';
-    const baseBranch = 'master';
-    const newBranch = `upload-${namespace}-${Date.now()}`;
-    const filePath = `namespaces/${namespace}/${file.name}`;
-
-    // Read the file and encode it in Base64
-    const reader = new FileReader();
-    reader.onload = async function(e) {
-        const content = e.target.result;
-        const base64Content = btoa(content);
-
-        try {
-            // Step 1: Create a new branch
-            const baseBranchUrl = `https://api.github.com/repos/${repo}/git/refs/heads/${baseBranch}`;
-            const baseBranchResponse = await fetch(baseBranchUrl, {
-                method: 'GET',
-                headers: headers
-            });
-
-            if (!baseBranchResponse.ok) {
-                throw new Error(`Error fetching base branch: ${baseBranchResponse.status} - ${baseBranchResponse.statusText}`);
-            }
-
-            const baseBranchData = await baseBranchResponse.json();
-            const newBranchUrl = `https://api.github.com/repos/${repo}/git/refs`;
-            const newBranchData = {
-                ref: `refs/heads/${newBranch}`,
-                sha: baseBranchData.object.sha
-            };
-
-            const newBranchResponse = await fetch(newBranchUrl, {
-                method: 'POST',
-                headers: headers,
-                body: JSON.stringify(newBranchData)
-            });
-
-            if (!newBranchResponse.ok) {
-                throw new Error(`Error creating new branch: ${newBranchResponse.status} - ${newBranchResponse.statusText}`);
-            }
-
-            // Step 2: Check if the file exists and get its SHA and content
-            const fileUrl = `https://api.github.com/repos/${repo}/contents/${filePath}?ref=${newBranch}`;
-            const fileResponse = await fetch(fileUrl, {
-                method: 'GET',
-                headers: headers
-            });
-
-            let fileSha = null;
-            if (fileResponse.ok) {
-                const fileData = await fileResponse.json();
-                fileSha = fileData.sha;
```

Comment:
Why are we setting the fileSha to the sha of the remote JSON response and not the local file? I am not super familiar with this github upload api schema (i.e. the sha requirement) so wasn't sure I am missing something

Reply:
Removed this. Doesn't seem to be necessary